dbt/include/snowflake/macros/materializations/incremental.sql

```diff
@@ -80,7 +80,9 @@
   {%- set unique_key = config.get('unique_key') -%}
   {% set incremental_strategy = config.get('incremental_strategy') or 'default' %}
   {% set tmp_relation_type = dbt_snowflake_get_tmp_relation_type(incremental_strategy, unique_key, language) %}
-  {% set tmp_relation = make_temp_relation(this).incorporate(type=tmp_relation_type) %}
+
+  {% set tmp_relation_suffix = '__dbt_tmp' if not model.batch else '__dbt_tmp_' ~ model.batch.id %}
```

Comment:
In order to leverage the work done in https://github.com/dbt-labs/dbt-adapters/pull/361, we'll need to cut a new dbt-adapters release + bump the dbt-adapters minimum dependency in dbt-snowflake.

I'm happy to do that refactor as a follow-on item to decouple those releases. Up to you @mikealfare

Reply:
Do we need this if we bump the dep on `dbt-adapters`? I think this would happen automatically with that bump. I'm sure I'm missing something.

Reply:
We don't, which is what I'm getting at. I can proceed with this change + remove it later once dbt-adapters have been released, or wait for the release and remove the `tmp_relation_suffix` construction here.

Reply:
We released the patched version of `dbt-adapters`, so let's remove this, per internal conversation.